src/test/java/uk/gov/hmcts/reform/hmc/helper/hmi/ListingMapperTest.java

```diff
@@ -237,4 +243,328 @@ void shouldReturnListingWithNoRoleTypeWhenEmpty() {
         assertNull(listing.getListingEndDate());
         assertNull(listing.getListingJohTiers());
     }
+
+    @Test
+    void shouldReturnListingForMultiDayHearingDurationIs720() {
```

Comment:
This and the other new tests look really useful in purpose, though are a little hard to read because of all the unrelated setup and assertions. Maybe we should move out the common setup of things that **have** to be done to a private method with `duration` as an argument, and then for these new tests only assert the multiday related fields? Since for the purpose of this test there are only really a few things of relevance that we haven't covered in other tests - something like:

```
HearingDetails hearingDetails = buildHearingDetails(duration);

Listing listing = listingMapper.getListing(hearingDetails, Collections.singletonList(HEARING_CHANNEL));

assertEquals(0, listing.getListingMultiDay().getWeeks());
assertEquals(2, listing.getListingMultiDay().getDays());
assertEquals(0, listing.getListingMultiDay().getHours());
```

Reply:
changed